editor/plugins/script_editor_plugin.cpp

```diff
@@ -2058,65 +2138,10 @@ bool ScriptEditor::edit(const RES &p_resource, int p_line, int p_col, bool p_gra
 	}
 
 	if ((debugger->get_dump_stack_script() != p_resource || debugger->get_debug_with_external_editor()) &&
-			p_resource->get_path().is_resource_file() &&
-			p_resource->get_class_name() != StringName("VisualScript") &&
 			bool(EditorSettings::get_singleton()->get("text_editor/external/use_external_editor"))) {
-
-		String path = EditorSettings::get_singleton()->get("text_editor/external/exec_path");
-		String flags = EditorSettings::get_singleton()->get("text_editor/external/exec_flags");
-
-		List<String> args;
-		bool has_file_flag = false;
-		String script_path = ProjectSettings::get_singleton()->globalize_path(p_resource->get_path());
-
-		if (flags.size()) {
-			String project_path = ProjectSettings::get_singleton()->get_resource_path();
-
-			flags = flags.replacen("{line}", itos(p_line > 0 ? p_line : 0));
-			flags = flags.replacen("{col}", itos(p_col));
-			flags = flags.strip_edges().replace("\\\\", "\\");
-
-			int from = 0;
-			int num_chars = 0;
-			bool inside_quotes = false;
-
-			for (int i = 0; i < flags.size(); i++) {
-
-				if (flags[i] == '"' && (!i || flags[i - 1] != '\\')) {
-
-					if (!inside_quotes) {
-						from++;
-					}
-					inside_quotes = !inside_quotes;
-
-				} else if (flags[i] == '\0' || (!inside_quotes && flags[i] == ' ')) {
-
-					String arg = flags.substr(from, num_chars);
-					if (arg.find("{file}") != -1) {
-						has_file_flag = true;
-					}
-
-					// do path replacement here, else there will be issues with spaces and quotes
-					arg = arg.replacen("{project}", project_path);
-					arg = arg.replacen("{file}", script_path);
-					args.push_back(arg);
-
-					from = i + 1;
-					num_chars = 0;
-				} else {
-					num_chars++;
-				}
-			}
-		}
-
-		// Default to passing script path if no {file} flag is specified.
-		if (!has_file_flag) {
-			args.push_back(script_path);
+		if (open_script_in_external_editor(p_resource, p_line, p_col)) {
+			return true;
```

Comment:
Should return `false` here.

Reply:
Why? Opening the file was a success. `edit()` should return true, no? If opening the file fails, then we proceed to try to open in in the internal editor, who will then decide what to return.

Reply:
`edit()` should only return true if it was successful in opening the file internally. eg.

```
if (open_script_in_external_editor(p_resource, p_line, p_col)) {
    return false;
}

/* Internal opening code */
```

Reply:
Doesn't seem very clear to me that that's the expected behavior, so I added some documentation clearing it up.

Reply:
By the way, the only place I can find that uses this return value is in `scipt_goto_method`, and the only place that uses _that_ return value is in `connections_dialog.cpp:775`. As I understand, it makes sure that the script is open in the internal editor before it selects a method connected via GUI.

Would this feature be desirable in the external editor as well?